idaes/apps/grid_integration/bidder.py

```diff
@@ -1291,3 +1291,395 @@ def _record_bids(self, bids, date, hour, **kwargs):
         self.bids_result_list.append(pd.concat(df_list))
 
         return
+
+
+class ParametrizedBidder(AbstractBidder):
+    """
+    Create a parameterized bidder.
+    Bid the resource at different prices.
+    """
+
+    def __init__(
+        self,
+        bidding_model_object,
+        day_ahead_horizon,
+        real_time_horizon,
+        solver,
+        forecaster,
+    ):
+        """
+        Arguments:
+            bidding_model_object: the model object for bidding
+
+            day_ahead_horizon: number of time periods in the day-ahead bidding problem
+
+            real_time_horizon: number of time periods in the real-time bidding problem
+
+            solver: a Pyomo mathematical programming solver object
+        """
+        self.bidding_model_object = bidding_model_object
+        self.day_ahead_horizon = day_ahead_horizon
+        self.real_time_horizon = real_time_horizon
+        self.solver = solver
+        self.forecaster = forecaster
+
+        self.n_scenario = 1  # there must be a n_scenario attribute in this class
+        self._check_inputs()
+
+        self.generator = self.bidding_model_object.model_data.gen_name
+        self.bids_result_list = []
+
+    @property
+    def generator(self):
+        return self._generator
+
+    @generator.setter
+    def generator(self, name):
+        self._generator = name
+
+    def formulate_DA_bidding_problem(self):
+        """
+        No need to formulate a DA bidding problem here.
+
+        Arguments:
+            None
+
+        Returns:
+            None
+        """
+        pass
+
+    def formulate_RT_bidding_problem(self):
+        """
+        No need to formulate a RT bidding problem here.
+
+        Arguments:
+            None
+
+        Returns:
+            None
+        """
+        pass
+
+    def compute_day_ahead_bids(self, date, hour=0):
+        raise NotImplementedError
+
+    def compute_real_time_bids(
+        self,
+        date,
+        hour,
+        realized_day_ahead_prices,
+        realized_day_ahead_dispatches,
+        tracker_profile,
+    ):
+        raise NotImplementedError
+
+    def update_day_ahead_model(self, **kwargs):
+        """
+        No need to update the RT bidding problem here.
+
+        Arguments:
+            None
+
+        Returns:
+            None
+        """
+        pass
+
+    def update_real_time_model(self, **kwargs):
+        """
+        No need to update the RT bidding problem here.
+
+        Arguments:
+            None
+
+        Returns:
+            None
+        """
+        pass
+
+    def record_bids(self, bids, model, date, hour, market):
+        """
+        This function records the bids and the details in the underlying bidding model.
+
+        Arguments:
+            bids: the obtained bids for this date.
```

Comment:
What is the specific data structure for a bid? Is this a list? Please provide a little more information in the comments

Reply:
Type hinting would be a good idea. E.g. the method signature could be:

```
def record_bids(self, bids: list, model, date, hour, market):
```

Reply:
I added variable types in code comments.